diff.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+
+
+def compare_source_files_objects(obj1, obj2):
+    diff_funcs = []
+    if obj1['name'] == obj2['name']:
+        if obj1['functions'] != obj2['functions'] or obj1['coverage'] != obj2['coverage']:
+            if obj1['functions'] == obj2['functions']:
+                obj1['functions'] = []
+            if obj1['functions'] != obj2['functions']:
+                for func1 in obj1['functions']:
+                    for func2 in obj2['functions']:
+                        if func1['name'] == func2['name']:
+                            if func1['exec'] is False and func2['exec'] is True:
+                                diff_funcs.append(func2)
+                obj1['functions'] = diff_funcs
+            list_val = []
```

Comment:
No need to use `list_val`, you can use `obj2['coverage']` directly.

Reply:
I thought it is not safe to modify values in the loop in Python; therefore, I've decided to make a list, where I put resulting values?

Reply:
It should be safe to modify the values of the list, it isn't safe to reduce/increase the size of a list.
E.g. this isn't safe:
```
l = [3, 4, 5]
for i in l:
    l.append(i)
```

The best option here would actually be to use a list comprehension. Something like this:
```
def diff_line(i, j):
    ...

obj2['coverage'] = [diff_line(i, j) for i, j in zip(obj1['coverage'], obj2['coverage'])]
```

Reply:
Ok, thanks for the explanation. Changed it now. :)